File: examples/tasks/stateless/hook.io/addition.py
import json,random

CLEAR_ALL = [True,None]

userMsg = json.loads(Hook['params']['d']) if 'd' in Hook['params'] else [0,0]
state = json.loads(Hook['params']['s']) if 's' in Hook['params'] else None

def send(stap,state=None):
	print('state=%s;'%json.dumps(state))
	print("recv(%s);"%json.dumps(stap))

if state:
	send( [ CLEAR_ALL,
			'Correct' if state==userMsg[2] else 'Incorrect',
			['Next'] ] )
else:
	x,y = random.randint(1,10),random.randint(1,10)
	send( [ CLEAR_ALL, x, '+', y, ['Answer',0,{'eN':1}] ], x+y )

File: examples/tasks/stateless/nanoscale.io/addition.js
function send(o,s){
  if(s!==undefined)response.body='state='+JSON.stringify(s)+';recv('+JSON.stringify(o)+')';
  else response.body = 'recv('+JSON.stringify(o)+')';
}

function recv(){
  if(request.params.d)return JSON.parse(request.params.d);
  return [];
}

function randInt(min, max){return Math.floor(Math.random()*(max-min+1))+min;}

if(request.params.s){
  send( [ ['',null],
          (request.params.s === (''+recv()[2]))?'Correct':'Incorrect', ['Next'] ],
          null );
}else{
  var x=randInt(1,10), y=randInt(1,10);
  send( [ ['',null], x, '+', y, ['Answer',0,{eN:1}] ], x+y );
}

